### ISSUE TITLE
Subscriber import improvements

### ISSUE THREAD

**Issue by ErisDS** [TryGhost/Ghost#6957]
This issue is meant as a collecting ground for some tasks we can do to improve the subscriber import feature.
## CSV-specific fixes:

In short, we need to be able to import from tools like Campaign Monitor. I tested a few imports and they failed. Firstly because there were quote marks around the email addresses and secondly because the header is "Email Address" which we weren't matching.

Therefore we needed to swap to use a CSV parser, which would fix #6865 as well and to update how headers were processed. 

This was all done in #6923 (I started writing this issue a while ago...)
## UI Improvements

Currently, the UI shows an upload progress bar, and then does nothing between the time when the file finishes uploading and the server finishes processing the email addresses and responds. The simplest fix here is to show a spinner and a phrase like "Importing subscribers" whilst waiting for the server.

Longer term, it might be better if the server returns after receiving the update, with some data about how many items are going to be processed (can we easily tell how many lines are in a file?!) and an id that can be used to poll the server for more info.

Server side we'd need to keep a reference to that id, and provide an endpoint that the frontend can poll to find out how the processing is going - e.g. that would return how many items have been processed and how many are left. 

I'm not sure if there are simple ways to do this quickly and then we can iterate longer term into having something like an actual job queue.. maybe.
## Performance improvements

Currently, uploading a mid-sized file can cause the server to take a long time to respond and/or crash. I've tried with ~7k addresses and it is very slow, ~350k addresses causes a crash.

Ideally we need to try to improve performance so that it's possible to import large subscriber lists successfully. Again, we could go so far as to have a job queue, but that would be more than what is required for a beta feature. 

Short term, we need to:
- test the limitations of the current implementation of csvparser in #6923. This version collects up all the information and processes it at the end, meaning a lot of data is stored in memory. 
- should revisit the `Promise.all` logic, and look at using `Promise.map` with a concurrency setting for adding subscribers to the DB.

In summary:

Short term tasks:
- [x] swap readline for a csv parser (implemented in #6923) 
- [x] fix handling of headers which contain "email" (implemented in #6923) 
- [ ] add a spinner to the Ghost Admin UI whilst the server is responding
- [ ] test the limits of loading all the data at once in the read-csv utility and maybe refactor this
- [ ] refactor Promise.all logic for importing subscribers so we can import larger files

Long term tasks:
- [ ] investigate if we can split the import response into an initial response, and then have an endpoint to poll 
- [ ] think about adding some sort of queue

**Comment by ErisDS** [comment]
Closing this with the `later` tag for when we start working on subscriber type features again after the launch of 1.0
